modules/navier_stokes/src/fvbcs/INSFVFluxBC.C

```diff
@@ -0,0 +1,40 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSFVFluxBC.h"
+#include "SystemBase.h"
+#include "Assembly.h"
+#include "MooseVariableBase.h"
+
+InputParameters
+INSFVFluxBC::validParams()
+{
+  auto params = FVFluxBC::validParams();
+  params += INSFVMomentumResidualObject::validParams();
+  return params;
+}
+
+INSFVFluxBC::INSFVFluxBC(const InputParameters & params)
+  : FVFluxBC(params), INSFVMomentumResidualObject(*this)
+{
+}
+
+void
+INSFVFluxBC::processResidual(const ADReal & residual)
+{
+  const auto * const elem = (_face_type == FaceInfo::VarFaceNeighbors::ELEM)
+                                ? &_face_info->elem()
+                                : _face_info->neighborPtr();
+  const auto dof_index = elem->dof_number(_sys.number(), _var.number(), 0);
+
+  if (_fe_problem.currentlyComputingJacobian())
```

Comment:
You raise a good point but I think my inclination would be to keep this as-is. This is from `FVBoundaryCondition.h`:
```c++
  /// Reference to SubProblem
  SubProblem & _subproblem;

  /// Reference to FEProblemBase
  FEProblemBase & _fe_problem;
```
`_problem` offers no distinction from `_subproblem` and I think that `_fv_problem` is a little obfuscating because the actual class type is `FEProblemBase`. Maybe someday we will alias `FEProblemBase` or something. 

In anticipation of a possible follow-on question, I'll note that `_subproblem` can either be a `FEProblemBase` or a `DisplacedProblem`.

Reply:
In my most recent commit I actually went ahead and created aliases for `FVProblem` and `FVProblemBase` and then I changed this data member name to `_fv_problem`. I liked the suggestion!